cds_ils/importer/documents/importer.py

```diff
@@ -0,0 +1,110 @@
+        self.update_document_fields = update_document_fields
+
+    def _set_record_import_source(self, record_dict):
+        record_dict["created_by"] = {"type": "import",
+                                     "value": self.metadata_provider}
+
+    def _before_create(self):
+        cleaned = deepcopy(self.json_data)
+        for field in self.helper_metadata_fields:
+            if field in cleaned:
+                del cleaned[field]
+        # save the import source
+        self._set_record_import_source(cleaned)
+        return cleaned
+
+    def create_document(self):
+        """Create a new record from dump."""
+        cleaned_json = self._before_create()
+        document_class = current_app_ils.document_record_cls
+        # Reserve record identifier, create record and recid pid in one
+        # operation.
+        record_uuid = uuid.uuid4()
+        provider = DocumentIdProvider.create(
+            object_type="rec",
+            object_uuid=record_uuid,
+        )
+        cleaned_json["pid"] = provider.pid.pid_value
+
+        try:
+            document = document_class.create(cleaned_json, record_uuid)
+            db.session.commit()
+            return document
+        except IlsValidationError as e:
+            click.secho("Field: {}".format(e.errors[0].res["field"]), fg="red")
+            click.secho(e.original_exception.message, fg="red")
+            # raise e TODO handle the incorrect records in the logging
+
+    def _update_field_identifiers(self, matched_document):
+        """Update isbns of a given document."""
+        existing_identifiers = matched_document['identifiers']
+        new_identifiers = [elem for elem in self.json_data['identifiers']
+                           if elem not in existing_identifiers]
+
+        return existing_identifiers + new_identifiers
+
+    def update_document(self, matched_document):
+        for field in self.update_document_fields:
+            update_field_method = getattr(self,
+                                          '_update_field_{}'.format(field),
+                                          None)
+            if update_field_method:
+                matched_document[field] = update_field_method(matched_document)
+            else:
+                matched_document[field] = self.json_data[field]
+
+        matched_document.commit()
+
+    def search_for_matching_documents(self):
+
+        isbn_list = [identifier['value'] for identifier in
+                     self.json_data["identifiers"] if
+                     identifier['scheme'] == 'ISBN']
+
+        doi_list = [identifier['value'] for identifier in
+                    self.json_data["identifiers"] if
+                    identifier['scheme'] == 'DOI']
+
+        authors = [author['full_name'] for author in
+                   self.json_data.get('authors', [])]
+
+        matches = []
+
+        # check by isbn first
+        for isbn in isbn_list:
+            results = search_documents_by_isbn(isbn)
```

Comment:
first 10/15 only, is it ok?
Same below...

Reply:
fixed all occurences